app/src/main/java/com/kamron/pogoiv/ScreenGrabber.java

```diff
@@ -58,6 +61,8 @@ public static ScreenGrabber getInstance() {
     @TargetApi(Build.VERSION_CODES.LOLLIPOP)
     public void exit() {
         if (projection != null) {
+            virtualDisplay.release();
```

Comment:
Clearing fields of the instance which is going to be GCed is should be useless (under some conditions), but I'm doing it for consistency. Releasing resources is different.

Finalizers are a parody of destructors and aren't invoked timely (Google `finalizers bad` for info), and `VirtualDisplay` doesn't even have a finalizer. Yes, Java is worse than C++ here. (Better-working alternatives include a `Thread` and a `ReferenceQueue`).

https://developer.android.com/reference/android/hardware/display/VirtualDisplay.html#release()
> Because a virtual display renders to a surface provided by the application, it will be released automatically when the process terminates and all remaining windows on it will be forcibly removed. However, you should also explicitly call release() when you're done with it.